validator.go

```diff
@@ -281,12 +280,11 @@ func (mv *Validator) Valid(v interface{}, tag string) error {
 		// unknown tag found, give up.
 		return err
 	}
-	state := &validateState{
-		path:      make([]byte, 0, 20),
-		pathStack: make([]int, 0, 10),
-	}
+	state := getState()
 	validate(sv, state)
-	if err := state.finalError(); err != nil {
+	err = state.finalError()
```

Comment:
Same as above. Is there a reason for this to happen before the `putState` call?

Reply:
as @tealeg says.